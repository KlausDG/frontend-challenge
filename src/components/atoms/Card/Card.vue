<script setup lang="ts">
import { Stack, Typography } from '..'

defineProps<{
  title?: string
  subtitle?: string
}>()
</script>

<template>
  <Stack direction="column" :spacing="16" class="card__container">
    <Stack direction="column" :spacing="4">
      <Typography>{{ title }}</Typography>

      <slot name="subtitle">
        <Typography variant="semibold">
          {{ subtitle }}
        </Typography>
      </slot>
    </Stack>
    <slot />
  </Stack>
</template>

<style scoped lang="scss">
.card__container {
  box-shadow: 0px 4px 8px 0px #00000014;
  border-radius: 8px;
  min-width: 328px;
  padding: 16px;
}
</style>
