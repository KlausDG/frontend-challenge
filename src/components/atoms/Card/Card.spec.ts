import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import Card from './Card.vue'

describe('Card.vue', () => {
  it('renders the title prop correctly', () => {
    const wrapper = mount(Card, {
      props: {
        title: 'Card Title',
      },
    })

    const typography = wrapper.findComponent({ name: 'Typography' })
    expect(typography.text()).toBe('Card Title')
  })

  it('renders the subtitle prop inside Typography when slot is not provided', () => {
    const wrapper = mount(Card, {
      props: {
        subtitle: 'Card Subtitle',
      },
    })

    const typographies = wrapper.findAllComponents({ name: 'Typography' })
    expect(typographies[1].text()).toBe('Card Subtitle')
  })

  it('renders the subtitle slot instead of the subtitle prop', () => {
    const wrapper = mount(Card, {
      props: {
        subtitle: 'Card Subtitle',
      },
      slots: {
        subtitle: '<span class="custom-subtitle">Custom Subtitle</span>',
      },
    })

    const customSubtitle = wrapper.find('.custom-subtitle')
    expect(customSubtitle.exists()).toBe(true)
    expect(customSubtitle.text()).toBe('Custom Subtitle')
  })

  it('renders default slot content', () => {
    const wrapper = mount(Card, {
      slots: {
        default: '<div class="slot-content">Slot Content</div>',
      },
    })

    const slotContent = wrapper.find('.slot-content')
    expect(slotContent.exists()).toBe(true)
    expect(slotContent.text()).toBe('Slot Content')
  })

  it('applies the correct CSS class on the container', () => {
    const wrapper = mount(Card)
    expect(wrapper.classes()).toContain('card__container')
  })

  it('matches the component snapshot', () => {
    const wrapper = mount(Card, {
      props: {
        title: 'Snapshot Title',
        subtitle: 'Snapshot Subtitle',
      },
      slots: {
        default: '<div>Snapshot Slot</div>',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
