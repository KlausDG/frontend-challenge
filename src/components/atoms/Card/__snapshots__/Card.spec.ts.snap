// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card.vue > matches the component snapshot 1`] = `
"<div data-v-69f42c1a="" data-v-9bb7b3b9="" class="stack__container card__container" style="flex-direction: column; gap: 16px;">
  <div data-v-69f42c1a="" data-v-9bb7b3b9="" class="stack__container" style="flex-direction: column; gap: 4px;">
    <p data-v-9bb7b3b9="" class="typography _caption_d0cbd2 _text_d0cbd2">Snapshot Title</p>
    <p data-v-9bb7b3b9="" class="typography _semibold_d0cbd2 _text_d0cbd2">Snapshot Subtitle</p>
  </div>
  <div>Snapshot Slot</div>
</div>"
`;
